### ISSUE TITLE
[BUG] Local cluster fleet-agent bootstrap missing `apiServerCA` that allows registration with Fleet

### ISSUE THREAD

**Issue by RenneNiku** [rancher/rancher#42018]
**Rancher Server Setup**

Rancher installed to kubernetes cluster (RKE2 1.25.9+rke2r1) with helm chart:

    helm upgrade -i --version 2.7.5  rancher rancher-stable/rancher -n cattle-system -f values.yaml

values.yaml has a couple changed variables compared to default:

    hostname: "rancherserver.our.domain"
    ingres.ingressClassName: "nginx"
    proxy: "our-proxy"
    tls: external

`rancherserver.our.domain` is reached via nginx-proxy, proxy resides outside Kubernetes cluster.

**Information about the Cluster**
- Kubernetes version:
  RKE2 1.25.9+rke2r1
- Cluster Type (Local/Downstream):
   Local cluster

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  Admin (on UI side)

**Describe the bug**

We're interesred in setting up Fleet.. and it seems that the
helm-command above also installed Fleet (so no need to install helm
separately using helm as describe here
https://fleet.rancher.io/installation.. or is there?)

We only got one cluster, hence operating on 'fleet-local'. After
setting up git repo, the cluster (under Continuous Delivery) stays in
'Wait Check-In' state.

After cheking few logs, this one pops out:

    kubectl logs pod/fleet-agent-855f8b5c9c-8pq8j -n cattle-fleet-local-system

    time="2023-07-04T09:45:06Z" level=error msg="Failed to register agent: looking up secret cattle-fleet-local-system/fleet-agent-bootstrap: Post \"https://rancherserver.our.domain/apis/fleet.cattle.io/v1alpha1/namespaces/fleet-local/clusterregistrations\": tls: failed to verify certificate: x509: certificate signed by unknown authority"

When checking how `fleet-agent-bootstrap` looks like, `apiServerURL` was set
to base64 encoded string of `https://rancherserver.our.domain` but
`apiServerCA` was empty. After taking CA from the  proxy server and adding
it as base64-encoded into `fleet-agent-bootstrap`:

    kubectl edit secret fleet-agent-bootstrap -n cattle-fleet-local-system

    apiServerCA: "LS...."

the error 'failed to verify certificate' went away and soon Rancher UI
showed everything to be in 'Active' state and Fleet was noticing changes
in git repo

Then, after a while, `fleet-agent` apparently restarted and we were facing
the same situation again and again editing the `fleet-agent-bootstrap` -secret on the fly
and adding `apiSeverCA` fixed everything.

**To Reproduce**
In our environmennt happens every time after installing  Rancher with helm the way show above.
Guessing that as Fleet's apiServerURL gets set to hostname which is actually alias for nginx-proxy-server  (which has self-signed certificates), to produce, one has to have proxy with self-signed certs and connect to Rancher via that proxy.

**Result**

In short, not having proper `apiServerCA` set, Fleet keeps giving 'failed to verify certificate' errors and not working properly.

**Expected Result**

There would be a possibility to set `apiServerCA` (or any other Fleet configuration value) for Fleet during Rancher installation via Helm.

**Question**

However, it is truly unpractical to keep editing the `fleet-agent-bootstrap` -secret every so
often, so the question is 

- how to set the `apiServerCA` (or agentCheckInterval or any other Fleet chart configuration variable) for Fleet
during Rancher installation. 

Could not see anything Fleet-related  in Rancher Helm chart documentation or values.yaml -file (apparently Fleet
somehow gets the `apiServerURL` from `hostname` set in  Rancher chart's `values.yaml`).

**Comment by Oats87** [comment]
@manno perhaps you have some insight here.

**Comment by Oats87** [comment]
Actually, rereading the issue this seems like it may just be a misconfiguration of the installation.

Rancher installs Fleet for you so it can manage its own resources. This is expected, so you should not need to manually install Fleet on any cluster managed by Rancher (including the local).

If your Rancher URL is secured using a private CA, you will need to provide a copy of your CA certificate chain to Rancher using the steps in the documentation: https://ranchermanager.docs.rancher.com/getting-started/installation-and-upgrade/installation-references/helm-chart-options#external-tls-termination  See the note linked about `privateCA`

**Comment by RenneNiku** [comment]
Just to confirm I've understood the docs correctly:
- `privateCa` should be set to to true for Rancher installation
- Secret named `tls-ca`  should be created  in `cattle-system` namespace having proper CA
- (Guessing from the comment above) Fleet chart `apiServerCA` would then be properly populated with whatever `tls-ca` has?

**Comment by RenneNiku** [comment]
Addind `privateCA` and `tls-ca`secret did the trick and `apiServeCA` was properly populated.

**Comment by petkozma** [comment]
I'm still getting this issue even I had set ```privateCA: true
